### PR TITLE
DIT-620: mongo encryption on PII

### PR DESCRIPTION
**Pull request by gokyo**

### CODE REVIEW

**Comment by ed0906** [comment]
I'm guessing using JSONFormatters didnt work? how did that other project do it?

**Comment by gokyo** [comment]
@ed0906 I wanted to use the implicit formatters, but it looked more work at first sight.
See this example: https://github.com/hmrc/vat-registration/blob/master/app/models/api/VatScheme.scala

Since this feature is needed ASAP (as you might now already), I preferred to use the simplest approach.
We can always improve it later with a better approach.

File: app/uk/gov/hmrc/bindingtariffclassification/crypto/Crypto.scala [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    applyCrypto(c) { s: String => crypto.encrypt(PlainText(s)).value }
+  }
+
+  def decrypt(c: Case): Case = {
+    applyCrypto(c) { s: String => crypto.decrypt(Crypted(s)).value }
+  }
+
+  private def applyCrypto(c: Contact)(f: String => String): Contact = {
+    c.copy(
+      name = f(c.name),
```
Comment (ed0906) [inline app/uk/gov/hmrc/bindingtariffclassification/crypto/Crypto.scala]: Whats `f`? Where is it defined?

File: app/uk/gov/hmrc/bindingtariffclassification/service/CaseService.scala [hunk 2]
```diff
@@ -27,27 +28,37 @@ import scala.concurrent.Future
 
 @Singleton
 class CaseService @Inject()(caseRepository: CaseRepository,
+                            crypto: Crypto,
                             sequenceRepository: SequenceRepository,
                             eventService: EventService) {
 
   def insert(c: Case): Future[Case] = {
-    caseRepository.insert(c)
+    val encryptedCase = crypto.encrypt(c)
```
Comment (ed0906) [inline app/uk/gov/hmrc/bindingtariffclassification/service/CaseService.scala]: Should we not be doing encryption/decryption in the repository? Seems like its a repository concern to me

Comment (gokyo) [reply app/uk/gov/hmrc/bindingtariffclassification/service/CaseService.scala]: I did not want to add logic there...

**Pull request merged**